polyfill/lib/calendar.mjs

```diff
@@ -1825,18 +1830,30 @@ const nonIsoGeneralImpl = {
   },
   dateFromFields(fields, overflow) {
     const cache = new OneObjectCache();
+    if (this.helper && this.helper.calendarType !== 'lunisolar') {
+      const largestMonth = this.helper.monthsInYear(fields, cache);
+      resolveNonLunisolarMonth(fields, undefined, largestMonth);
+    }
```

Comment:
I wonder if we should move these into `helper.calendarToIsoDate` so that we don't have to check the `calendarType`? That'll make it easier in the future to refactor the calendar helpers to be actual classes, which is somewhere down later on my to-do list.

Reply:
In any case, `this.helper.calendarToIsoDate` is called unconditionally below here, so we don't need to check if `this.helper` is truthy.

Reply:
I don't think I understand the helper structure well enough to attempt that (and note that it would apply to both `calendarToIsoDate` and `monthDayFromFields`), but I did remove the unnecessary guard.